### PR TITLE
Reference fixes for #189

### PR DESCRIPTION
**Pull request by galdolber**
This is not intended to be use as is, as it has more changes than the main issues.

I found the same clojurescript commit affecting specter: https://github.com/nathanmarz/specter/issues/261

Not sure what is the best fix for reader/read-string*.

### CODE REVIEW

**Comment by Engelberg** [comment]
It seems like you've done a good bit of research on the new changes to Clojurescript, and how they affect instaparse. Can you summarize the problem for me, and what you did to fix it in this code?

**Comment by galdolber** [comment]
Hi Mark, I just found the commit that breaks the code and then followed what Mike Fikes suggested for specter here: https://github.com/nathanmarz/specter/issues/261.
I don't really understand why and how this is affecting instaparse. The commit(https://github.com/clojure/clojurescript/commit/9a8196ebfe4265feda88a06de84affb9df469012) is pretty simple, just handles a few special cases for performance.
Sorry I couldn't help more.

**Comment by galdolber** [comment]
This seems to take care of the error, but not sure why.

screen shot 2018-11-12 at 4 53 51 am
